src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1871,6 +1871,107 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node, insOpts instOptions)
+            //      {
+            //          throw new OverflowException();
+            //      }
+            //
+            //      Vector256<double> op1_f64 =
+            //          Vector256.ConvertToDouble(Vector256.WidenLower(Vector128.ToVector256Unsafe(op1))));
+            //      Vector256<double> op2_f64 =
+            //          Vector256.ConvertToDouble(Vector256.WidenLower(Vector128.ToVector256Unsafe(op2))));
+            //      Vector256<double> div_f64 = op1_f64 / op2_f64;
+            //      Vector256<long>   div_i64 = Vector256.ConvertToInt64(div_f64);
+            //      Vector128<int> div_i32 = Vector256.Narrow(div_i64.GetLower(), div_i64.GetUpper());
+            //      return div_i32;
+            regNumber op2Reg   = op2->GetRegNum();
+            regNumber tmpReg1  = internalRegisters.Extract(node, RBM_ALLFLOAT);
+            regNumber tmpReg2  = internalRegisters.Extract(node, RBM_ALLFLOAT);
+            emitAttr  typeSize = emitTypeSize(node->TypeGet());
+            noway_assert(typeSize == EA_16BYTE || typeSize == EA_32BYTE);
+            emitAttr divTypeSize = typeSize == EA_16BYTE ? EA_32BYTE : EA_64BYTE;
+
+            CORINFO_FIELD_HANDLE maxValueFld;
+            CORINFO_FIELD_HANDLE negOneFld;
+
+            if (typeSize == EA_16BYTE)
+            {
+                simd16_t maxValueIntVec;
+                maxValueIntVec.i32[0] = INT_MAX;
+                maxValueIntVec.i32[1] = INT_MAX;
+                maxValueIntVec.i32[2] = INT_MAX;
+                maxValueIntVec.i32[3] = INT_MAX;
+
+                simd16_t negOneIntVec;
+                negOneIntVec.i32[0] = -1;
+                negOneIntVec.i32[1] = -1;
+                negOneIntVec.i32[2] = -1;
+                negOneIntVec.i32[3] = -1;
+
+                maxValueFld = emit->emitSimd16Const(maxValueIntVec);
+                negOneFld   = emit->emitSimd16Const(negOneIntVec);
+            }
+            else
+            {
+                noway_assert(typeSize == EA_32BYTE);
+                simd32_t maxValueIntVec;
+                maxValueIntVec.i32[0] = INT_MAX;
+                maxValueIntVec.i32[1] = INT_MAX;
+                maxValueIntVec.i32[2] = INT_MAX;
+                maxValueIntVec.i32[3] = INT_MAX;
+                maxValueIntVec.i32[4] = INT_MAX;
+                maxValueIntVec.i32[5] = INT_MAX;
+                maxValueIntVec.i32[6] = INT_MAX;
+                maxValueIntVec.i32[7] = INT_MAX;
+
+                simd32_t negOneIntVec;
+                negOneIntVec.i32[0] = -1;
+                negOneIntVec.i32[1] = -1;
+                negOneIntVec.i32[2] = -1;
+                negOneIntVec.i32[3] = -1;
+                negOneIntVec.i32[4] = -1;
+                negOneIntVec.i32[5] = -1;
+                negOneIntVec.i32[6] = -1;
+                negOneIntVec.i32[7] = -1;
+
+                maxValueFld = emit->emitSimd32Const(maxValueIntVec);
+                negOneFld   = emit->emitSimd32Const(negOneIntVec);
+            }
```

Comment:
This could be simplified to a `simd_t vecCns = {};` and then a for loop for `elementCount` iterations (given `elementCount` = `simdSize / 4`).

Then doing `emitSimd16/32Cons` based on the `simdSize`.

Reply:
It might, alternatively, be cheaper to just check if the result of `divpd` is `2147483648`. That gives us a single comparison instead and is the only case that will be out of range for `int`.

Reply:
isn't `negOneIntVec` can be just `simd32_t::AllBitsSet` ?